From 96a3cee1dd4700a9e033f545db5891e4fa8c073d Mon Sep 17 00:00:00 2001
Message-Id: <96a3cee1dd4700a9e033f545db5891e4fa8c073d.1569024025.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Mon, 19 Aug 2019 02:26:19 -0400
Subject: [PATCH] promisor-remote.h: drop extern from function declaration
To: Git Mailing List <git@vger.kernel.org>
Cc: Junio C Hamano <gitster@pobox.com>,
    Christian Couder <christian.couder@gmail.com>

During the creation of this file, each time a new function declaration
was introduced, it included an `extern`. However, starting from
554544276a (*.[ch]: remove extern from function declarations using
spatch, 2019-04-29), we've been actively trying to prevent externs from
being used in function declarations because they're unnecessary.

Remove these spurious `extern`s.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---

This patch should be placed on the tip of cc/multi-promisor.

I wonder if there's any way we can automate this check...

 promisor-remote.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/promisor-remote.h b/promisor-remote.h
index 8200dfc940..c60aaa5cce 100644
--- a/promisor-remote.h
+++ b/promisor-remote.h
@@ -15,17 +15,17 @@ struct promisor_remote {
 	const char name[FLEX_ARRAY];
 };
 
-extern void promisor_remote_reinit(void);
-extern struct promisor_remote *promisor_remote_find(const char *remote_name);
-extern int has_promisor_remote(void);
-extern int promisor_remote_get_direct(struct repository *repo,
-				      const struct object_id *oids,
-				      int oid_nr);
+void promisor_remote_reinit(void);
+struct promisor_remote *promisor_remote_find(const char *remote_name);
+int has_promisor_remote(void);
+int promisor_remote_get_direct(struct repository *repo,
+			       const struct object_id *oids,
+			       int oid_nr);
 
 /*
  * This should be used only once from setup.c to set the value we got
  * from the extensions.partialclone config option.
  */
-extern void set_repository_format_partial_clone(char *partial_clone);
+void set_repository_format_partial_clone(char *partial_clone);
 
 #endif /* PROMISOR_REMOTE_H */
-- 
2.23.0.565.g1cc52d20df

